src/api/manager/job/syncInventory.ts

```diff
@@ -67,10 +68,22 @@ export async function runSyncInventory() {
       // If source was not found in response from API, always mark it as gone
       return { ...inventorySource, status: 'gone' } satisfies WithId<Source>;
     }
+    const statusUpdateCheck = () => {
+      const databaseStatus = inventorySource.status;
+      const apiStatus = apiSource.status;
+      if (apiStatus === 'new') {
+        return apiStatus;
+      } else if (databaseStatus === 'purge' && apiStatus === 'gone') {
+        return databaseStatus;
+      } else {
+        return apiStatus;
+      }
+    };
+
     // Keep all old fields from the inventory source (name, tags, id, audio_stream etc), but update the status
     return {
       ...inventorySource,
-      status: apiSource.status
+      status: statusUpdateCheck()
```

Comment:
IMO you should either move the `statusUpdateCheck` outside the `runSyncInventory` or just inline the functionality as creating a function and then directly calling it has little benefit.